test/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+# Copyright Louis Dionne 2016
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+
+# Each unit test is a small, self-contained CMake project with a `check` target
+# and possibly CTest targets. This might look a bit odd, because we're using
+# CMake to run CMake, but we do this instead of using `add_subdirectory` to
+# make sure each test is run in its own "sandbox".
+#
+# The METABENCH_UNIT_TEST_OPTIONS variable can be used to pass options down to
+# the unit tests. It is used e.g. in the Appveyor build.
+
+file(GLOB test_dirs
+    LIST_DIRECTORIES true
+    RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+    "*"
+)
+foreach(dir ${test_dirs})
+    if (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${dir}/CMakeLists.txt)
+        continue()
+    endif()
+
+    add_custom_target(test.${dir}
+        COMMAND ${CMAKE_COMMAND} -E remove_directory ${dir}/build
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${dir}/build
+        COMMAND ${CMAKE_COMMAND} -E chdir ${dir}/build ${CMAKE_COMMAND} .. -G${CMAKE_GENERATOR} ${METABENCH_UNIT_TEST_OPTIONS}
+        COMMAND ${CMAKE_COMMAND} --build ${dir}/build --target check
+        COMMAND ${CMAKE_COMMAND} -E chdir ${dir}/build ${CMAKE_CTEST_COMMAND} --output-on-failure
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        VERBATIM USES_TERMINAL
+    )
```

Comment:
Or did you mean why setup a `custom_target` and then call it from `add_test` as opposed to simply doing the work in `add_test`? If so, this is because `add_test` does not allow for multiple commands.